//const fs = require('fs')
//var reader = new FileReader();

let inp = '{{{{{<!>},<q!>oi<{!!uiuQ!ao{,q!!!!o,>,<>},{<{i>},{{},{{<e,Q!!!!!>}!!!!o!!!>!}!>,<!e!!!>!!!>}!>,<<!!q!>,<!!,>},{<,>}}}},{{<{<!!{!><!>,<Q,>,<!{!!!>i,u!>,<!!!>,<,aq>},{<{>,{{<!>},<!>},<i!!!{u!>},<e{!!!>}i,q!e!!!>},<!!!>,<>}}},{{<}<!!!>!}o<q{uii{a!!!!!>qu!{!!!!<>}}},{{{<!>,<QquoQa!!}>}}},{{{<ueu}!>Q!!!,>},<{{>},{{}},{}}},{},{{{{<!!>,{<uQ!>},<!!!>!!{>}},<!q,!u{!>o>},{{{<u!>,<!!!!<i!!!>!!!>i!>},<u!!{!>!!!>,<{Q>}}}},{{},{{<!>,<}i!>Q!!!!{}>},{<,!!!>!<!!!>>}}}},{{{{{{{{<{!!q!>},<!o{!i!!qo,>},{{{{<!!!>q!>},<<!Q>,{<!!o>}}},{<a!!!!,!>,<!!!!}e,o,!>},<{!!!>uq>,{<!>,<}!!}!>,<q!>,<{!!!>},<!!!e!!,{iQ>}}}}},{}},{<>}},{<qq!>!>!!!>qi!!!>q!>}u!>},<i!>,<!!!>,>,{<!!!>,q!>,<<!!qo!!!>ao,!!!>!>!!aao>}},{{<!>!>,<!>},<,o!!!>},<eQ>},<i!>},<e,!!!>,<}qQ!>},<!!!>}!e!!>}},{<,Qea>,<!>},<a{>},{<Q!!!>eu!{a,>,{<{!!e<q!!e!!!!a>}}},{{{<>}},{{{}},<!!ia!!<!>,<!!!!,!>,<q{!!>}},{{<!!!>,!!!><ou!,,!<,,e{!!ue>,{{}}}},{{{<Q!!Q!>i!>!!}!!!>Q!>},<!!!>q<ioo!!>}},{{<!!!>},<<e!>},<q<!>,<!>},<o>},<}Q,o!!!>,<!>!eu!>!>},<!>},<!!!>e},oii>},{{<aq!!!>!>!<uo!!QQ>},{<uqqeo!o<Q!>a,>,{<!,ua!>!!!>!>},<ao>}}}}},{{{},<}!>aQq<aeo!!!!!!<!>},<i!>},<>}},{{},{{<a{qqq!>,<!o!!u,e!!!>!!!>!>},<>,{}},{{<Q!>},<a!!i!!uQ!>,<{>}}}},{{{{{{<!>,<qi!!,ou,oo!a}q!!!a!!{aaui>}},{}},{}},{{{<!!o<!,!>,<!Q!>},<!!{,>}}},{{{<Q!!aQa!>,<,!>},<!!!>>,{<!!!>,<!a{!>},<>}},{<}a,!}!>},<!>Q!o!!!iq!e!!!>a<!a!!e>},{{<{,u!!<uQ<a{Q}u!>},<,!>},<<!>},<!>},<>},<}u!!<i!!!>,<e!>,<!!!>!>!>,<ao!!!a!>,<Q!!!<>}},{<<aq!!!><>,<}>},{{<q!!e,!>,<!>!>},<e,oo!!!!!>!!!>!!!>Q!<o!>},<i>,{{}}},<uQ>}},{{{<!!iu!!a>}},{{{{},{<!!!>!!!>},<>}}},{{{{{<}!!!>!!!>>}},<e!>},<!>},<<o{!>},<<{<a{q>},<,!!!e{<!>,<QQ!}!<>}},{{}}},{}}},{{{<{<!>},<}>}},{},{{<>,{{{{<<qi!!!>qQ!>{<>}}}}}}},{{{{<!!!>{!!Qui}!>,<{aq{!!!>!<!!!>i{!!}o>},<!>},<{Q!{!>,<Q!!u,u}!!!>,<!>,<!}!!}>},{<QQ!!!>{uu!!!!!>}!>},<!!<ei>,{<Q!>},<i>}},{{<}!!!>uQq!!>}}},{{<!!!>},<!aa!>},<a>,{<!!{!!a>}},{}},{{<qa!!u{!!!!,!!<>,<!!!!ii}qi!>},<,>},{{<u!>!!}!!o!>},<e{!>a<e>,{{}}},{{<!!!>!>,!!!>!!!>},<!q!>,<o<Q<!!!>!>,<>},{{},<!!<!>},<i{{ii!!!>!>,<<<>}}},{{},{{<!!!>},<>}},{{<a}Q!!u!>,<!i},a{{<Q{{Q>},{{{<!>},<<!!}e!!!>,!!!>a{>}}},{{{{<a,!>},<!>,<!!,e!!q{!uiq>},{{<!>},<!>q<<uQ<!!!>!>!!!>!!>}}},{{<u!!}qai!!}!!!>!>,<!>},<u!>,q>}},{<<qq!e,!!!>},<iu<!>,<!>,<>}},{{<{!i!!<}euu!!i<!!!>!!iq!!}!>},<a!o!!!>},<>}},{{<aeuuia!>,<!>},<ia{,o!>,<!!!!!>>}}}}}}},{{{{},{{<e!>},<Q<u{e!!!>,!>},<a>,<!>},<>},{{},{<!!o,a!>},<!!!>,<{>}}},{{{{{{{{{{<!><e!!!!!>!!ee!!!!o<!>io!!!!!>!>>},{<}>}}},{<u!>,<o}!!oa!>!>},<!}}!!}!!!>,!!!>e!>!ee>,{<q!!e{oqq,Q{eQ!>},<>}}}},{{},{{<q!a!>o<i!>},<!o<!!!>{>}}},{{<!>!u!>!!!>o!>,<u!>,<,e>},{<}i{u{!!{!>},<a!!e!>},<!!!u>}}},{{<!>!>,<>,<}!oe!>,<!>,<uuea!Q!!a>},{{<u!>q<!>,<}!>},<>},{<a!Qq{!>,<!!},!q!>u!!a,!!>,{}}}},{{{<uoQ!!!!qe{i>},{}}}},{{{<}!oo!!!>!!!>{!!!>q>},{{<!>},<u!!!>,<!>,<!!>},{{},<a}Q!!!>>}}},{{<<!>},<!>,<o!a<q!><!!!!,!!!>!>uQ!>>,{<q{i>}},{<e!>,<Q{!>},<!!!{Q<!>,<>},{<!!!>,<!!Q!!!><,eoi,ue{!!!!}}!!,u!!!>}>,<!!ia<!!!>{!>},<!>,<q}>}},{}},{{{{},{<{!>,<!!!!oua!>},<,!>},<o<<i!!Q{>,<}a<ui,o>},{}},{{{<!!e{{!!e>},<!!!!!!!}uQquQ>},{<a>}},{}},{{<!!!>!<<,!>!!Q{,{!!!i}!>,<{>,{}},{{{{<!!!>},<ue!!<!>!>,<!!!>,<!!!>!!!Q{!>,<e!>o!!!>,o>},{<<QQiu>}}},{{<!uo!>},<!>},<o!!!>q>},{{{<}!!,!!!><o!>,Qo,>}},<!!!>{{e!!!>>},{{<!>},<o>,<e}!>au!!!!q!>},<Q!>q!>},<Q!>Q{!!!>},<!>},<{>},{{<<ai!!},!!!>uu!>,<Q!ou>},{{{<}!!ieqq}!!u!>},<!!!!!q,a<qQ!>,<Q>}}}}}}},{}},{{{<!>!!u!!oa!>},<{!!!>o!>oa!!!>!!u!!!>q!q>,<eu}Qqi!<!!o{>}},{<!!o>,<!q!!!!i>},{{},{<ee!!!>,<Qe>}}},{{{},{<e}u},{!>!!ie{!>},<!!!>a!,q>}}}}},{{},{{<!!Qui!!e{!!!!!>}!>,<!>},<!!!!}e,!>>}}},{{{<!>},<!!!!,!Qoia!!!ia}!!quu>},{<!>,Q!}!,!>,<e{!>},<!!!!oi{Qa>}}}},{},{{<!!!>QQi!!o{u!>a!>!!q!!{{>}}},{{<!>!!!>!!,e}!>!!Qqo!!!>,<}{!u!!!>!i>,<,e<Q!>,<!!!<!>e!!{{aq>},{{{<!!q!>,<!{!>q!!a}>}},{<o,}!>eaa!>},<!iua>}},{{<!>,}<uoa!!e!!!q!!!>!qq,ei!!{>}}}}},{{{{<!!!!{!Q}qo!>},<!!o{!!!>>},<!>},<!!ue}!>},<}<e!!o!>},<!!Q>}},{<!>},<>,<!!i!!aa!>,<!u!>},<>},{{<Qoaqa!}!!!i!!q,!>},<o!>},<!>e!>},<>}}},{{},{{<!!iq!>,<!!u}ei!!oq!!a>},{}}},{{{}},{{<!!i!!!>{,}<,a!!a!!!>!>{!o>},<!!!>{o>}}}}}},{{{{{<!!}!>,<!!!>},<{a{{u!>!>!>},<!!e!>,<u>},{{{}}}}},{{{<<e>,<QQ!>},<!>,<!>e>},{<>}},{<!!,ea!!!!<!!!!>},{{<i!!!>uQ<ueoi!>},<,e}QQ!>},<a>},<>}}},{{{{{<!<!!oQ!>>}}},{},{<o!>,<}q,!>,<a!!{!!!>!!!i!>},<,i<!e>,<}e!>,<!!!>!!!!o!>e!u>}},{{{<uq{!!!>oo!>},<!!!q!!u>,<iee!>!ia!!oq>},{<{!iq!>,<aaq{iQo!!i!qee>},{<<!!!>,<!!uo!!!!iia}!!!>,u!!o{!>,<!{!Q>,<!!q!>!!!u}{iqou!>},<!ioia!>},<{!>>}},{{<}i!>},<!!!>!!,{!>},<,o!!!>,<,!!e<!>,<!!!>,u>,<>},{},{{{<!!!>Q}{>},{{<!>,<i}aQii!!!>!>u}u{Q{!!i>},<u,!Qq!!!>!>a!Qqu>}},{{<Q}qoa!!{!>!!<qi}aQ!q!>,!oq!!a>,<,!u!>,<a!>,<Q!{!!!>!>},<{!!aQo>}}}}},{{{<oq!!u!!i!!Qe}a}Q!>},<!!!!,{>}},{},{<!>!!!,u!!e!!!>!>!>},<!!!>>,{<!>},<q!>!>},<<!>,<>}}}},{},{{{{{{<!!oqq!>},<,,!!!!e!>,<>}},{{<<!><!>!!!>!e!!!!{,Qeae!,{iiii>,<u<>},{{<!!e>},<>},{{<,{!ou{ei>}}},{{<i>},{<a!>,<!>,<!!o<!>},<u<e!>,<>}}},{{{<aQ<!!!!e<qo!>,<u!!!>!>},<!>uq!!<!!!>>},{{},{<}<,!!!>u<!>},<}!>!,>,<<!!!!{!>},<!>,<!i{<!!!!!>},<>}}},{{<ea!!!>!}!!<!!e!!o,!>q>}}}}},{{{<iQoo<oee>},{{},<u}!!!>!>a!>,<!>,<}<!!!oq!!aQ!>},<,,>}},{{{<!>},<o,{!qae!>},<QQ!!!>}>},{<Q,!!a<!>!!,!>a>}},{},{<i!!Q}!>!>>,<Qq!>!>},<,Q>}},{{{}},{<<,o!>,<>,<!o!!!>!>,<>},{}},{{{{<,!!!>,!i!i,!o>},{<!{!q}a!!!>!e!>!!,eQq!>},<Q{!><e>,<!>{!!!>i{!!Q!ae<Q{}!>,<Q!>!>},<>}},{{<!!!>i!>i!!!q}!>,<Q>},{<!>},<!!!!!><,qui<!!!>!!!!!>q!q!!e!oQo>}}},{{<Q!>,<u{oa!>,<<!>>}}}}}},{{{{<!>!{,>,{<a!>},<{<{!>,<q!>},<i!>>}}}},{{{<!!o<q!,<>,<!!!>a>}},{{<!>},<oa}eu}ee,!!,Q}!q!>,<!!ia>,{<}}a{i!>!!}!!,<!!!>,<>}},{{{}},{<}!>,ea!u},Q{!!!>i!>,<!>},<!>},<!,}>}},{<!!!!!>u!>},<},}Qe!>,<!!!>,<<!!ie}!!!>},<>,{}}},{{{},{{<!!!>oQ}!!,!>euq!!<!>,<!>},<}!!!>!>},<!>,<a>},{<!!!!!>u<,!u!i!>!>!!!!!i!o>}}},{<>},{}},{{{{{<!>!!!>!!u{Q!!!>,<>},<,<q!ua!!<!!!>Q!!u!>,<!e}u!>},<>},{<!>,<!i!!>}}},{{{{<!!!><!!i>},{{<{ia,<!!i!a<,e<i!!oQo!!!!<!!u!>a>,{<a!>,<}{o!!ie!>},<iqe<!>,<o!>,<!!}}<>}},{<,!!!>ee!!!>e{}},u!Qe>}}}},{{{<{equq>},<!!{!>,<<!!!>e!>oaaQ!>},<}!!!>,<Q>},{{{<u!!!>!>},<!>,u>,<!>,<!!{!!Q<!>,<}}!!q>}},<o}}iq!!Q!>!>a!!!!!!!><>}},{{{<}!!!>,<!>!>,<!!!>!>,u{<o>}},{<e!q!>,<,!>,<e!>,<{Q!>,<<q!!o>}}}}},{{{{},{<!!!>>}},{{}},{{{{<!!!>o!!{!>,<a,e,!!!>,<i>,{<aQi!!!!,<!>,<!!e>}},{<!!!>},<}!!Q<,!!!>},<e{ei!!q!>,<!>,<Q!!!!Qio!>,<>}},{{<i!>!!!>{!!<!!uqq}oia!>,<>}}},{{},{}},{<qa!>},<!!!!!>!q!>qau!q{>,<{<e!>e!!!>}!>,,>}}},{{<o!,a<{,!!!>!!,q}qi!>},<{<q!u>,{{<}<q!!!>!i!!io,,!>},<Q!>},<>}}},{<e!>o!!o<!>},<{>,{<!!!!!>!>},<!o<!>,<!!>}}},{{{<!>,<<}!!Q>,<Q!{uo!>!!!!{,!>},<euqq!!!>,>},{},{<<e!!o}>,<e<u!>,<uQ!!i>}},{{{},{{{}},{<aQ!!i!>,<!>i>,{{}}}}},{{{<!!!!!>!!!>,<!!{!!u!>!>Q<qu>},{{}}},{{<u!!!>!!!>Q!!!>>}}},{{},{<}!<ui!>,<<q>,{}},{{},{{},<q{>}}},{}},{{<>},{{<a,!>},<,,e{a!>},<!!!>u!!!!!!!>qi{<,}>},{}},{{<q!!!<aui<e!>,<u!!!{!>,<<}!>,<>},<!>o<!!i!>!!!!>}},{{<<!!ui!!,!aQ!!!!oQ>}}}}},{{{{{},{{{<!{>}},{{{<qQeo!>o!>},<!ee!!!!!!,<o!!<!>},<!>,<u!!,>}},<ii!!!>!!a}>},{<!>,<!>},<a<Q!!!!aeua!e,!!>,{<,aq}{QQi!!}u}!!!!!>!!!!q!!{>,{<!!!!Q!>},<!!!>aau>}}}},{{},{{<a!!!>!>},<eQe{!i!>>},<!>,<!>i!!!>u}o!!!>!,!>},<>},{<!!!>,!>},<!>},<q!>!,!!,{q!>,<!!!>,<!>!!!!!>,<e>}}},{{{{{}},{{<qo!!!>o!!!>a,e>},{{{{<o!>,<>,<{{>},<<!>,<!!!>!>!>,<}>}}},{<Qu!>},<o!>,<,!>},<o!!!!!>,<!>},<!}q!!,o!>!<!!}>}},{{{{<qu,!>}}e!!!>!!q>},{<,>},{<!>,<}{!>,<!!!>,<o!q!!i,!!!>!!!!>,{<uu{!!!>!<!!!>!!!>Q{!!eQq<u<o}>}}},{{<e!>},<iue{i>,{<,uaQ>}},{{{{<aeQ!!!>},<,!!!<!>o>}}}},{{<oe!!<>}}},{{<o{>},{{{{<ai!>,<!!!>},<q}!!!>!>u!!!>q>},<!>,<Q!>,<{!!!>!>},<{!>,<>},<QQ<e!>},<!i!>},<eo>}},{{<!>},<!!!>},<!!!>o!>},<!>,<!!Q!>>},{{{{<a},!>oa{Qa,>}}},{{<q!!!>,!>},<!!!>},<oQ{!>},<,,uaee!aQ>},<>}}}}},{{{<Q<Qu,i>,<!!e{!!,!!}<>},{<i!>},<<!!!Q!!!!!!o!!,i!!!>ia!>},<{!!,>}},{{{<}{!!qq!ea<}{!!!>Q!!!!!>!>},<!!ao>,{<<ii!>,<QQ!>!!!!!>},<!!eQe{Q!>,<>}},{{<!>},<!a}!!!>i{!>},<a!!!,>},{{<}!>},<!Q<<!!!>>},{{}}}},{{},{<!!!!Qa<QQq{!!!>!>!!>,<!eiqea!>,!>},<<ai>}}},{{<}e!>!!Qo!>},<Qqoe>},<}!>},<<}!!!>Qqq>},{{}}}},{{{<q!!a{!!!><!>},<>,{<!!,!>e,!>!u{i!>,<>}}},{{<!>},<q!>},<>}},{}},{{{},{<q!!}}<o!!,{,!>},<!!!>!!oq>}},{}}},{{<!!{>},{{{<e,u!>},<,,o!>,<!!ee!!!!ai!>,<!>},<Q>}},<e!>,<<,!>,<!>{ia!a!}!!!>o!!e!!!>!!{!e>},{{<!!!!<>}}}},{{{<{>,{<Q!!<!}}!!{Q!!!<i!>},<u!!{>}}},{{}},{<,!>,<!{>}},{{<>},{{},{{{<oo!!!!q<<}a<!,!>,<!a!!!>,<<>},{<e!>,<e!!!!!>!!!>Q!!q!!!>i}>,<!!q!!!!!<q!!u!>},<!>},<!},!a{!>},<o>}},{{<!>},<}e}>,{}}},{{{{{<>},<i!>io,!ui}e!!}ii>}},{{{{}},{<q!!}a>}}}},{},{{{<!>,<e!>,<!>uu!!,!>},<>},<!!!<!!!!!>{<!q!!o!Qooiu!qi}!!!>!>,<uu>}}}}},{<i!!!>,<!o>}},{{{<,!{{e!>},<!!!!{,<}{<!!!>Qai<qo>},{}},{},{{<q,!>},<!>},<!>!>},<!>,<i!!!!ieo!>>,<oo}{!!!>o!!!>a!>,<q!>!u,!}}{!!!q!>!!!>>}}}},{{{},{{},{{{{<!!>},{{}}}},{{<!>,<{!>}<<}!!!!}!>>},{}}}},{{{<!>,<<>},<u{}!>,<q,{!u,!><Q!>},<iq{>},{{},{<>}},{{},<!>,<q!>!!<!{!>!>q{!>{{>}}},{{{<o}Qi<!!!>!>,<!!!!q!!a!>},<!>!>},<o!!Q>},{{{<i!>},<!,!!q!!!>,<u,i<i<Q!>,<<u!>,<}>}}}},{{},<!!!>}!>},<a!>},<i,}!{!!o>},{{<!!!!!>!!!>!q{!!!!u>},{<}e!>,<!!{>}}},{{{<!!!>},<!oi!>},<!>,<>},{{<!!u!uo!!!>!!i{!>eqi!>},<q>,{}}},{{{}},{<!Q>}}},{},{{<u!>!u,!!!>,<}aee!>,<}e!!!>q!!>},{{},{<i!!Qi!!{a!Qua!!Q}o!!!><ao>}}}},{{{<Qaq!>},<qa!!ie,iQ>},{}},{<>}}},{{{{},{{}}},{{{{<!!!!!o!>,<!!!>,<e!>},<!!{!!!!!>!!!>},<u!Q!oQa,>},<!>,<!>,<!!!>,<<!>!!!>e>},{{{{<}qQ{}q!!Qq!!!>u>},<aq!!<!>,<,!>,<u!>},<o!!uo{!a!>>}}},{{{{<!>},<,!>,<!!!>{!>uaua!!<!!e>},{{}}},{{<!!q>},{<!>,<<!!,!!!>!o!!>,{<e}eo>}}},{{{<!!!>{>}},{<ao>,<!!ue!>},<}oQ!>,<>},{<,eaq!>},<!>,<>,{<i!!u!>},<!>,<!!!!!>o!>,<a!>},<u!>e!>u!!!>!a!!!>>,{<q!!o!!<Q!!{o!>!!!>!>},<i!!q!<<!>a}i>}}}}},{{{<!qQiou!>},<!!>}}}}},{{{},<>},{<qq>}}}},{{{{<!}{i,!!<Q!!u!<!<!!}!>,<>,<ooqo!!!!!>!!q!!o!>},<!>},<q>}},{{<q!uua!>!>,<,ou!>},<>,<i!!!><!>},<!>},<aQ!>,<!!!>,<!!>}}},{{{<ie!!!!a>},<Qi<!q>},{{}},{{<>}}},{{{}}},{{{}},{{{<}<!!!>,<!!!>i!>,<<!!!>>},{<!>,<o!!o!!!>,<{!!ouqqi!!!>Q<<!!<<{o!>,<>}},{{<!!a,>},<,uu>}},{{{}},{<!!!>!!o!!!>}Qi!!<{o!!!>!i!>},<i>}}}},{{{{{}},<!<Q!>ua!>,<e!>aa!!!!e}u!!!>,<i>},{{}}},{{},{{<!>,<!!!!!>,!o!!eu!>,<{!!o,,!!!>>},{{}}},{}},{{{{<q!>,<eqeq,!>},<<!>,<aa<>}},{{},{{<e{>},{<!,ie!!!>!>},<!>,<!!,!,!>!qo>}}}},{{{<e!>!!!>!uu!!!>u!>,<o>}},{{},{<!!>,<!>!>},<!>},<!>},<iia<a!!>}}},{{{<!!!!!>,<i!!<!>qq>},<u!!!>},<!o}!>,<QQ>},{{},{<<aua!!e!!!!!>!>a!!!!{!!!>e,!!!>qi>}},{<}aQo<!!!>>}},{{{<<!>,<>},<!>,<<!!!>},<!>,<}!!!>,<q>}}}},{{{{}},{{<!!!><!!!!!>u!}!!u!!i>},{{}},{{{{},<ou!!qi!>!>},<oeq!>},<Q!ou!!qo>},{<!!!!!>>,{{<!!!>,<!i<,!>},<{{!}u!!!><!!!>i!!!>q!>q>},<i!!!>!ae!>>}},{{}}}}}},{{<q,>},{<!!!>}!!ae!!!u,<e>,<<oqe!!uaoqQ<>},{<a{!!oe!!!>!uQ!!!>!!a!>},<>,<!!i!>},<qi,!!}!>},<!u!!!>,<!!!>!>,<u!>},<>}},{{{},<!!<,>}}},{{{{<uo<>}},{{<<!!oeieuq<!!!>>}},{<!ae{!!!>,,iuioi!i<u!!!>!!!>i!>>,{}}},{{{<ua<!>,<Q>},<!!!>e!>,<,{Qi<!!!>!!!>!>}{}Q!!e>},{{},{}},{{<!!!>a!>!>,<!a!>},<eu!!i!u!e!i,e!>},<!>e!,a>,{{<!>},<,!!!>!>,!!{io!>!>},<!aq>}}},{{{},<{{{<Q{!!!>{{,a!!!>>},{{{<,e!>},<>},<!!!>!>,<!><!>!}!!!>!!Q!!!>!>},<!!!!!!i!>!!!>}!>},<!{>},<>}}}},{{{{{{{<!>,<u!>,<o!>o!!!>!Q,!>eu<ueQqa>},{{},{<!!!!!>,!!!>e,,}oq!!!>!!{!!>}},{<!!<!a!>!!!}ui!!!>},<i!aQou!!!>,<>}},{<{!>!!!!!>},<>,<i!!!>Q>},{}},{{<o,{!!<Qu!>!>i{!>,<{!!!!!>{!>!!!>},<>,{}},{<>}},{{{<q>},{<!!eu!>},<uoo!>!!!!!>!>!!!>i>}},{{<!>!>,<!o!i{!>,<}>,{{<!!,o,!!!>>}}}},{<q>,{<e<!q!u>}}}},{{<!>u!!!><!>,<!>i}>}}}},{{},{{}},{{<qi!e!!!!!>!>!!!>{!>,<i!!!>io!>u>},<!!!!!>u,a!>},<!>}!!!>!!o!>,<!><}>}},{{}}}}},{{{{<e<!>,<iQie>,<i!>,<,!!a!>,<,!>},<!!e!>{u!>!>,<{>},<,i!,u!!!>u>}}}},{{},{{{<!>,<a<!!Q}q!>,<>}},{},{{<!>!>},<!!!>,e>},<!!!>},<ee>}}}},{{{<iQQ!!{!!!>a<ea!,!!!>,<>,{<!!u!!ei!><<!q!>,<!{!!!ao<<!>!>,<a!<>}},{{{<oo!>,<>}}}},{{{{}}}},{{{<>}},{{}},{<!>},<!>,<a>,{{}}}},{{{{<u!>,<>,{<o}},!>,<o!>,<<!>},<iu>}},{}},{<!!!!!!!>e!>},<!>},<>}},{<>,{{<!!<o!!!>!,{,Qi!!}!>!>{>}}},{<,!!!>},<{q,e!>,<oQ!>,<<}u!!>,{<>}}}},{{{{<!!!>!!!>!a!>!!!>!>>,<}aQuq!!!!i!!!>},<i>},{},{{{<aQo!!,!!!>a!!!!ue!{o!<i!>,<e,uoe>}},{{},<!!q!>,<a!!{{q{e!!,ea!>,<i!!!>,ia>},{{<<!>},<{>},{}}}}},{{},{{{{{{<oqa!!!!!>!,q!>,<q!>},<!>!a!!!>e>}}}}},{<qQ!{Q}!!!>{}q!!!!!!!!,!ie>,<!>,<!>},<!!!!o{>}},{<{>,<{!Q{!!!>>}},{{{<!>},<ioo!!!>},<!!!>},<>}},{{{},{<i!!uQ!>},<!>},<>,{<}!>},<!ei!>},<aeq!>},<!>},<!!!,>}}}}}}},{{{<qu!>!>},<,!!u!!!!,!>},<!>},<,!>},<!!{!iQe>,{{<e!!}!>,<!!q!!<>},<!}!!!>!!ou!!!!Q>}}},{{{{},{<ieqQq}{!>,<a}}}aQqQq!!!>>}},{{{<!!!>!!Qeeu!>!>},<!!QQ<Qqi>},{{{{<a!!!>!}Q{!>,<oQeQ!>},<>},{}},{<,!!!>},a!>,<!>,<o!>o}!>},<<{a{e>}},{<e,!!!>},<<!!!!!,!>e!,!!a!Q>,{<{!>},<Q!>!o,a>}}}},{}},{<!!!qQe<>}},{{<o{,!!!>},<!!!!!>,{u!!!>a!>,<>,{}},{{},{}}}},{{},{<,,!>!>a,!!!>i!!iu!!!!!!!!!!!!!>Q>}},{{}}}},{{{{{},{},{{<Q!>{o!Q!,q!>},<ui}!>,<,>}}}},{{<!>},<q,{!>i!!q!!!>i>}}},{{{{{<!>},<Q!>!>!!!>,<!>},<>},{{<ui!<}!!!>uu{{<!{u!>,<!>},<!u!>!>,<>}},{{<e{,oQ{o!!!>e!!!>}!!}!>},<!!,,!>,<o!!u>}}},{{{{<!>},<!>},<{!!Q!>i!!q!>},<i!!,!!!>io>}},<!!Qu!!!>!!i>},{<,{!!!>!!i!!!!!>!!}q,Q!>},<o}!>},<qo>,<oq<!>,<e<!!!i<!q!!!!!>>},{{{<!!!!<>}},<uq!!!>!!!>},<{}ao!>>}}},{{{<!!e!!!>!>,<!>!!i>}}},{{<{},!!oo{uu!!!>,<Q>},{<uoq!!>},{{<i{!>},<<!>>},{{{{{<!!!>uu!>!>{>}},{{<{>}}},{},{{{}}}},{<>,<!>,<!!!>!!!>},<!!i<<{aoi>}},{{<e!i!!!!!>!>u!!o{!!aiq>},{{}}}}},{{{{<!!!>aQ>,<!uQ{<>},<!>},<oo!!!>ioQoeio!!i!!!>,<a!!a!!!>,<Q<}>}},{{{{{<!>!!!>},<!>,<!!u{i{i!!<i,!!QQQ>}},{<a!!,,!!!!e!!e>}},{<a!!!>>},{{<!>i!>oq{q!o!>},<!!!>!>!!q!!i!}>},{<,a!>},<!>},<q!!!!!!a!!!>>}}}},{{{<a<!!!>!!,>}}},{{<!>,<u!!!>!eu}!!!>}!!!>,!!!>ao!>>},{}}}},{},{}}},{{{{{{{{<!{!!!><}qoe!aq!>!au!>},<!!q>}},{}}},{{<!i,eQ!>,<ie!>},<aaQ}e<<>}}},{{{},<q,!!o!Qa!!eiai<!,},>}},{{}},{{{{<q<{!!!!!!eu!uqq!>u!!!>!<>},<iQ>},{{<ea>,<!>aua{!e!!!>!>},<e{!!!!Q!>,<>},{{<}o!!e!>o!QQ,e!Q>,<o<!>,<<!!!!!ua!>!!!>q!>},<!>,<!!u!>},<!>}!!>},{}},{{<!!{a>}}},{{<!!!>!!a{!q}!>,<<!!!>!>,<Q,!>Q>,{}},{<!!!>auQq,o!!!>!!Q}!>},<oe!>},<!!}u!!!>,<o>}}}}},{{{{{<!,!u!>},<<<!>!oi,<!>,<{!!!>!!!>!!>},{<!!e}<!!i<{io!>,<<u!>,<a!!,qqq!>,<>}},<i!}!q!>!!uu}!!<!>,<,e>},{{{},{}},{}},{{<<}!!!!,Q!!!!!>!!o!!i,!!o<{>}}},{{<oqoo!,}!!!u!>q!>!>,<Q>},{{<}}!!!>>,<<!ie!{,u!>},<!!!>e!>>},{{<!!iu!>},<}>},<!>>}},{{{<!!!>,<!oui}}Q>}},<ui>}},{{{{{},{<qo{!>},<Q!!}!!!>{>}},<!>},<!!!!!>!>},<,eo!!a,q!!!>!>},<}!!{!>,<!!,!>,<i>}},{{<q!!}!!!>!<!!!>},<>,<!!!>qq!!q>},{{<eQu!!,Q!>!>,<aQ!euu>,<!!!>Qaiuu!!,!>,<o!!<!u{!>},<q!>,<Q!>>},{<!<aai<}!!o!>,<!!a}!!!>o!!!}a>,<!!!>!!,!!!>!!!>e!!!>>}}},{<ia{>}},{{{<qqqQ,>,<!>,<!>,<!!!!eu<aieqi,}{!!!>!!!!!iQ!!!>!>>},{<!!q!>QQ{!!!>},<>},{{<!!!>,>},<}!!Q!!!>!!!!!>a!!!>e}Q!!o,}q>}}}},{{{{<!>,<}!>!>},<!!}}}!q!q!>,<q}au!!!>!>,<!!!>>}}},{{<<!!!>oaq,!!<q}i}<q!!!>!>,<!>},<>},{{{<!!!>!>!>},<!>}!!!>,,i<Q!!!>,!!!>{>},<!},>},{{<}i!!}q{{oaaQQ<}i>},{<!,!>},<<<eqq}!!!>>}}},{{<!!ou}>},{<!>>,{<{!!u>}},{{<e!!!>}!!!!!>!>},<!!Qa!>},<aq!>!!!>Qo!!!!!>,o>,<!a{i>}}}},{{},{<ia!>,<!>o!>},<aa,i{<{,,}}!>>},{{{<a!!!>{>},{<a>,{<q!>},<!!u{>}}},{<!>,<>,{}}}},{{{},<!qa>},{{{<}!!!>Qa!!oi!a!>},<!!!!ae<>,{{}}},{<!}<!>!!!>!!!>!aioi,!>>}},{{{<>},{}},{{<<{!!e!>!>},<a!e<>},{{{<ei<!>!!!>a>}}}}},{{},<a!!!>{o,<}a!!!>},<>}},{{<!!{{!!!>!!!!!>!>o!!!>i!>},<>}}}}},{{{{<!>!e!>},<!ioui,!>,<u}>,{}}},{{<!!!>ei!>},<,>},{<!{QQioo!>,<q<,q<!>},<q!!!>a!!!!!!!>!>},<u>,<{o!>,<<}iQe>},{{<o!!Q!Qa!>,<}!{!>!!!!u!>,<q>},{<!!!>eq!!!>},<}!!!>!!!>uuu{!!!!!>},<{,Q!,>}}}},{{<}!>!>i,!!!!!>!>>},{{},{<!!!!o!!!>Qi!>},<>}},{{{{<,!{!!!!!>},<qo}!!o>,{<{}!!a,!{>}},<{<}!!!a>},{<,q!!aouq!><{>}},{<aoq}>}}},{{{<!!a!!!>!!!>!>},<},u!!Q!>},<<e!!}Q!a!!i!!!>a>},{<i!!!>i,q!>},<u<{a!u!!!!!>},<}Q>}},{{{},{<o!>!>},<!!!>,<{e{!{a!!!>,<!!!!!>{!!!>,<!!!>},<!>},<>}},{{},<q!q!>,<}a<,{qeaa>},{<uoQqioe<!q,a>}}},{{{<{e!qe!!{!!!!a!>uu!>{>,{}}}}},{{{{{{{<!}Q<>}},{}},<!!<u<}!!!!!>,<qii!>,<!!!>},<Q!!Q!!uo>},{<!!ua!oo}aq!!>,{<!>},<!},{!>},<qo!>,<u!!!>Q{!>},<<o>}},{<a!!!>aa!>},<o>,{<>}}},{{{{{{<!>,<>},{<}!>qu!!,!!!a!!!>>}},<!{!!Q!!,,o!!!><uo!!Q<Qaio!!!>},<>}},{<Q<i!>},<!!i>,{{{{}}}}}},{<!Qq{!>{{>,{<q!!!u}!o!{>}},{}},{{{<,!!ao>},{<!>,<!>,<!ea!>,<!>},<!!!!!>,<}q!>,<!>},<!>>}},{{<Q},,!!o!>,<{!>,<!>,<{>}}},{{},{{{<!!!>,{>},{<>},{{},<!!!>!o!!!!!!Q!!!>!>,<!!i!!!>!!o<>}},{{<!>e!!!!!>}Q,!>,<Qa!!!>{uqua>}}},{{{}},{<!>},<,,{!!!>!>},<!>},<!>,<!>!{!!ai{Q!!!>a!!e!qo>}},{{<!>}i,>,<!>!!!>ue}e!>,<{!>},<i>},{{}}}}},{{{<!>},<,>,<!>},<!!!>!e!>},<ui!!>},{{{{<!!,o!,!>},<>},<o!!!>,<!!ie!}uQ<}}ae!!!>,!!e{>},<}Q!>},<!>!!!!!>!<>},<!!!!!!<!>,<>},{}},{{{<uuao!>!>,<!>}!!!!<!!!>e<!>},<!!!!i{e!!!!!>>},{{{<{!>},<!!!>>},{{}}},{<!>iQ,,>}},{<a!>},<q!>ii<!!q>,<e,i{!!!>!!{}!>}!!!e{<!!}!>},<!!!>},<Q{>}},{<}!>!!e!ue!!!><!!!>Qoa!!Qea!!!>},<!>,<>,<o!{!!!>Q!!{Q}e!!,!{!>},<!>,<!}>}}},{},{}},{{},{{{{},{}},{{{{<!>,<iQq{,>}},<,{o!>!>}qqo>},{<!!<!>,>}},{{{<Q,,a!,i!!!><!><!!!>}q!!!{Q!Q}!>,<!>},<>,<!!,!!uuoo!>i>},{<!!>}}}},{{{<Q!q,!!<!>,<!!}!!!!>},<!>},<QuQi!!!>,<!!i!>,<Qoi!>!!eQ!>!>},<}>},{{{{<!>>},{},{{},<!!!!qu!>},<a<{qi}!!!!!{!!a!>,<>}},{<e!!!>,!>!!e}<!>},<!>,<>,{}},{{{<!!!>!!!>!>Q>}},{}}},{{{}}}}},{{<!!!!!>iu!!Q!>},<}u<!>,<!>},<!!!>q<<e>},{<q>,<!i!!!>Qa!!!!o!!i>},{{<!!q!>,<q<aQ!>,<,o>},{{<!>,Q>},{}}}},{{{}}}}}}}';//reader.readAsText('day9.input.txt','utf-8'); //fs.readFileSync("./day9input").toString('utf-8').trim()

let garbage = false, score = 0, depth = 1, garbageCount = 0
for (let i = 0, c = inp[0]; i < inp.length; i++, c = inp[i]) {
  if (c == '!') i++
  else if (garbage && c != '>') garbageCount++
  else if (c == '<') garbage = true
  else if (c == '>') garbage = false
  else if (c == '{') score += depth++
  else if (c == '}') depth--
}